### PR TITLE
Treat expired auth errors as fatal 

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Review by 2hdddg — CHANGES_REQUESTED** [review]
Since every connection has a timestamp from when it was used you can use that in combination with a timestamp in the pool to know which connections to remove. You shouldn't need to tag every connection that way.

File: neo4j/internal/pool/pool.go [hunk 1]
```diff
@@ -414,3 +418,21 @@ func (p *Pool) Return(c db.Connection) {
 		p.log.Warnf(log.Pool, p.logId, "Server %s not found", serverName)
 	}
 }
+
```
Comment (2hdddg) [inline neo4j/internal/pool/pool.go]: Both of these are accessing structures written to from other threads, will not work.

**Comment by fbiville** [comment]
Changed the PR completely now @2hdddg, let me know if that approach is better.
I'll add more test coverage as soon as you think the approach is OK.

**Review by 2hdddg — APPROVED** [review]
That should do it!
Perhaps a test on bolt4 to ensure it. That should be enough since there is no extra logic needed in the pool

**Review by 2hdddg — APPROVED** [review]

**Pull request merged**